### PR TITLE
feature: Saga support jackson json parser

### PR DESCRIPTION
**Pull request by long187**
### Ⅰ. Describe what this PR did
Saga support jackson json parser

### Ⅳ. Describe how to verify it
```
client {
  rm {
    sagaJsonParser = jackson
  }
}
```
or 
```
client {
  rm {
    sagaJsonParser = fastjson
  }
}
```

### CODE REVIEW

File: saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java [hunk 1]
```diff
@@ -39,14 +45,31 @@
...
+
+
+    public DbStateMachineConfig() {
+        try {
+            Configuration configuration = ConfigurationFactory.getInstance();
+            if (configuration != null) {
+                this.rmReportSuccessEnable = configuration.getBoolean(ConfigurationKeys.CLIENT_REPORT_SUCCESS_ENABLE, DEFAULT_CLIENT_REPORT_SUCCESS_ENABLE);
+                this.sagaBranchRegisterEnable = ConfigurationFactory.getInstance().getBoolean(ConfigurationKeys.CLIENT_SAGA_BRANCH_REGISTER_ENABLE, DEFAULT_CLIENT_SAGA_BRANCH_REGISTER_ENABLE);
+                this.sagaJsonParser = ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.CLIENT_SAGA_JSON_PARSER, DEFAULT_SAGA_JSON_PARSER);
+            }
```
Comment (wangliang1986) [inline saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java:68]: When the configuration is null, the default value is not set.

**Comment by wangliang1986** [comment]
Would it be better to change Seata's JSON parser? Not just SAGA mode.

**Comment by a364176773** [comment]
please resolve code conflicts

**Comment by a364176773** [comment]
please resolve code conflicts

**Comment by long187** [comment]
> please resolve code conflicts

done

File: saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java [hunk 2]
```diff
@@ -39,14 +45,33 @@
...
+    private boolean sagaBranchRegisterEnable = DEFAULT_CLIENT_SAGA_BRANCH_REGISTER_ENABLE;
+    private String sagaJsonParser = DEFAULT_SAGA_JSON_PARSER;
+
+
+    public DbStateMachineConfig() {
+        try {
+            Configuration configuration = ConfigurationFactory.getInstance();
+            if (configuration != null) {
+                this.rmReportSuccessEnable = configuration.getBoolean(ConfigurationKeys.CLIENT_REPORT_SUCCESS_ENABLE, DEFAULT_CLIENT_REPORT_SUCCESS_ENABLE);
+                this.sagaBranchRegisterEnable = ConfigurationFactory.getInstance().getBoolean(ConfigurationKeys.CLIENT_SAGA_BRANCH_REGISTER_ENABLE, DEFAULT_CLIENT_SAGA_BRANCH_REGISTER_ENABLE);
```
Comment (wangliang1986) [inline saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java:66]: `ConfigurationFactory.getInstance()` to `configuration`

Comment (long187) [reply saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java]: done

**Comment by wangliang1986** [comment]
problem: In the springboot project `application.yml`, config the `seata.client.rm.saga-json-parser=jackson` is not valid, the value of the sagaJsonParser is still the default value `fastjson`.

**Pull request merged**